idea/idea-frontend-api/src/org/jetbrains/kotlin/idea/frontend/api/components/KtExpressionTypeProvider.kt

```diff
@@ -25,6 +28,9 @@ public interface KtExpressionTypeProviderMixIn : KtAnalysisSessionMixIn {
     public fun KtDeclaration.getReturnKtType(): KtType =
         analysisSession.expressionTypeProvider.getReturnTypeForKtDeclaration(this)
 
+    public fun KtFunction.getFunctionKtType(): KtType =
```

Comment:
Maybe `getFunctionalType` is a better name here. Also, the KDoc is missing. And looks like, the return type may be reduced to `KtFunctionalType`.

Reply:
Renamed to `getFunctionalType`, and add KDoc. Alas, we do `asKtType` conversion at the end, which doesn't have type reduction, so, for now, `KtType` is the best abstraction.